modules/apps/notification/notification-web/src/main/java/com/liferay/notification/web/internal/portlet/action/GetObjectFieldNotificationTemplateTermsMVCResourceCommand.java

```diff
@@ -96,19 +100,46 @@ protected void doServeResource(
 				));
 		}
 
+		JSONArray termsJSONArray = getTermsJSONArray(
+			_objectFieldLocalService.getObjectFields(
+				objectDefinition.getObjectDefinitionId()),
+			objectDefinition.getShortName(), themeDisplay);
+
+		List<NotificationTermProvider> notificationTermProviders =
+			_notificationTermProviderRegistry.getNotificationTermProviders(
+				objectDefinition.getClassName());
+
+		for (NotificationTermProvider notificationTermProvider :
+				notificationTermProviders) {
+
+			Map<String, String> notificationTerms =
+				notificationTermProvider.getNotificationTerms();
+
+			for (Map.Entry<String, String> entry :
+					notificationTerms.entrySet()) {
+
+				termsJSONArray.put(
+					JSONUtil.put(
+						"termLabel",
+						language.get(themeDisplay.getLocale(), entry.getKey())
+					).put(
+						"termName", entry.getValue()
+					));
```

Comment:
@gabrielwas I would try to reuse this from the `BaseNotificationTemplateTermsMVCResourceCommand` to avoid duplicate code

Reply:
It's a tiny piece, I'll keep it as it is. Thx for the suggestion.